### PR TITLE
Add tests for pids

### PR DESCRIPTION
**Pull request by gliptak**
Signed-off-by: Gábor Lipták <gliptak@gmail.com>

### CODE REVIEW

File: pids_test.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+		bytes,
+		defaultFilePerm,
+	); err != nil {
+		t.Error(err)
+		return
+	}
+	metrics := Metrics{}
+	err = pids.Stat("test", &metrics)
+	assert.NotNil(t, err)
+}
```
Comment (crosbymichael) [inline pids_test.go]: Can you `gofmt` this code?

Comment (gliptak) [reply pids_test.go]: updated

File: pids_test.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package cgroups
+
+import (
+	"encoding/hex"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/stretchr/testify/assert"
```
Comment (crosbymichael) [inline pids_test.go]: I don't think we use this package in containerd and definitely not this one.  Can we remove this?

You also need to use `goimports` to format the imports in this PR

File: pids_test.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+	if pids == nil {
+		t.Error("pids is nil")
+		return
+	}
+	resources := specs.LinuxResources{}
+	resources.Pids = &specs.LinuxPids{}
+	resources.Pids.Limit = int64(10)
+	err = pids.Create("test", &resources)
+	if err != nil {
+		t.Error(err)
```
Comment (crosbymichael) [inline pids_test.go]: You can use `t.Fatal` here so you don't have to have a `return` in every block

Comment (gliptak) [reply pids_test.go]: updated

File: pids_test.go (hunk 2 above)
Comment (gliptak) [reply pids_test.go]: updated

**Comment by gliptak** [comment]
@crosbymichael  are there other updates you would like to see?

**Comment by crosbymichael** [comment]
LGTM

Thanks for all the updates to this, the code looks really good!

**Pull request merged**